### ISSUE TITLE
"Literal[10]" cannot be assigned to type "Literal[1]"

### ISSUE THREAD

**Issue by solotim** [microsoft/pyright#1574]
**Describe the bug**
Pyright gives strange error.

**To Reproduce**
See below

**Expected behavior**
The code is correct

**Screenshots or Code**
If applicable, add screenshots or the text of the code (surrounded by triple back ticks) to help explain your problem.
```python
class test:                                                                              
    def __init__(self, x, y):                                                            
        self.x = x                                                                       
        self.y = y                                                                       
                                                                                         
                                                                                         
if __name__ == '__main__':                                                               
    a = test(1, 2)                                                                       
    b = test(3, 4)                                                                       
    a.x = 10                                                                             
#      Cannot assign member "x" for type "test"                                          
         Expression of type "Literal[10]" cannot be assigned to member "x" of class      
       "test"                                                                            
           "Literal[10]" cannot be assigned to type "Literal[1]"                         
       [Pyright: reportGeneralTypeIssues]                                                
```

**VS Code extension or command-line**
Running it with vim plugin Coc-Pyright

**Comment by erictraut** [comment]
Thanks for the bug report.

Here's why this is occurring. When Pylance encounters a class with no type annotations in its `__init__` method, it makes that class into a "pseudo-generic" class parameterized by the `__init__` method parameters. It's as if you had typed:

```python
class test(Generic[_T1, _T2]):
    def __init__(self, x: _T1, y: _T2):
        self.x = x
        self.y = y
```

This approach allows the type checker to make intelligent assumptions about the types of instance variables initialized in the constructor.

The problem here is that when you called the constructor with values of `1` and `2`, the TypeVar constraint solver assigned `_T1` the type `Literal[1]`. It should have widened the literal type to `int` before assigning it to `_T1`. That way, `x` will get the type `int`, and you can reassign it other `int` values such as `10`.

That's probably more information that you cared to know. :)

In any case, this will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.118, which I just published. It will also be included in the next release of Pylance.

**Comment by solotim** [comment]
Thanks a lot, erictraut. 
Good to know you have a quick fix. Yeah, the tech details are much verbose than I can understand. LOL. 
👍
